libs/community/langchain_community/embeddings/llamacpp.py

```diff
@@ -116,7 +117,14 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             List of embeddings, one for each text.
         """
         embeddings = self.client.create_embedding(texts)
-        return [list(map(float, e["embedding"])) for e in embeddings["data"]]
+        if not isinstance(embeddings["data"][0]["embedding"][0], list):
```

Comment:
is it possible for this line to raise IndexError where it previously would not? e.g., if `texts` is empty or contains empty strings.

Reply:
ok

Reply:
Sorry for the confusion. I didn't mean to *request* that we raise an error where it previously would not.

From an ignorant reading of the code (i.e., no knowledge of the behavior of the client) the update introduced the potential to raise IndexError where it previously wouldn't. I've pushed a change that passes your tests. Let me know if you see any issues with it.